### PR TITLE
EOS-25962: Attach policy to user

### PR DESCRIPTION
**Pull request by shrimetta**
Signed-off-by: Shri Metta <shri.metta@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: auth/server/src/main/java/com/seagates3/controller/UserPolicyController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	
+	private final UserPolicyResponseGenerator userPolicyResponseGenerator;
+
+	public UserPolicyController(Requestor requestor, Map<String, String> requestBody) {
+		super(requestor, requestBody);
+		userPolicyResponseGenerator = new UserPolicyResponseGenerator();
+	}
+	
+	public ServerResponse attach() throws DataAccessException {
+		LOGGER.info("--------Attach Start------------");
```
Comment (shalakadharap) [inline auth/server/src/main/java/com/seagates3/controller/UserPolicyController.java:25]: remove these loggers

File: auth/server/src/main/java/com/seagates3/response/formatter/AbstractResponseFormatter.java [hunk 2]
```diff
@@ -64,4 +64,10 @@ abstract ServerResponse formatGetResponse(
 
  public
   abstract ServerResponse formatChangePasswordResponse(String operation);
+
+ public
+  abstract ServerResponse formatAttachUserPolicyResponse(String operation);
```
Comment (shalakadharap) [inline auth/server/src/main/java/com/seagates3/response/formatter/AbstractResponseFormatter.java:69]: This is not required as AttachUserPolicy and DetachUserPolicy doesnt return any response back.

File: auth/server/src/main/java/com/seagates3/response/formatter/xml/XMLResponseFormatter.java [hunk 3]
```diff
@@ -177,6 +177,16 @@
       return success(operation);
     }
 
+    @Override public ServerResponse formatAttachUserPolicyResponse(
+        String operation) {
+      return success(operation);
+    }
```
Comment (shalakadharap) [inline auth/server/src/main/java/com/seagates3/response/formatter/xml/XMLResponseFormatter.java:183]: This is not required as AttachUserPolicy and DetachUserPolicy doesnt return any specific response back.
You just need to return 200 on success.

File: auth/server/src/main/resources/IAMroutes.json [hunk 4]
```diff
@@ -1,4 +1,6 @@
 {
+	"AttachUserPolicy": "UserPolicy#attach",
```
Comment (shalakadharap) [inline auth/server/src/main/resources/IAMroutes.json:2]: Check formatting. Keep it consistent with existing entries.

**Comment by shalakadharap** [comment]
Add UserPolicyImpl class which will actually perform/call ldap level operatons.
In your case, it wil be -
" ldapmodify to put policyId against user"

**Comment by shrimetta** [comment]
Closing this PR. Will create a new PR with updated changes

**Pull request closed without merging**